modules/history/src/history.js

```diff
@@ -0,0 +1,194 @@
+const self = require("../index.js");
+const historyHelper = require("./history-helper.js");
+
+/**
+ * History module allows you to manage an history for your project.
+ * It use data-store module to stack entities on 'snapshot' method
+ * and restore them using 'go' method
+ * History module can be use to implement system like Undo/Redo, Ctrl+Z / Ctrl+Y
+ */
+class History {
+
+    constructor() {
+        // max Length of history, above that oldest snapshots are deleted
+        this.maxLength = (self.app.config) ? self.app.config.get("maxLength") : 50;
+        this.pointer = -1;
+        this.snapshots = [];
+    }
+
+    /**
+     * Set Max amount of snapshots stored by the history.
+     *
+     * @param {Number} max Length of history, above that oldest snapshots are deleted
+     */
+    setMaxLength(maxLength) {
+        if (Number.isNaN(Number(maxLength)) || maxLength <= 0) {
+            throw new Error("Setter: History.setMaxLength parameters is not a number.");
+        }
+        this.maxLength = maxLength;
+    }
+
+    /**
+     * Max amount of snapshots stored by the history.
+     *
+     * @readOnly
+     * @return {Number} max Length of history, above that oldest snapshots are deleted
+     */
+    getMaxLength() {
+        return this.maxLength;
+    }
+
+    /**
+     * Currently stored snapshots count.
+     *
+     * @readOnly
+     * @return {Number} current history length
+     */
+    getLength() {
+        return this.snapshots.length;
+    }
+
+    /**
+     * Tell if current pointed element is first in history
+     * It means we can't go any further forward
+     * It means pointer is on last snapshot taken
+     *
+     * @readOnly
+     * @return {Boolean}
+     */
+    isFirst() {
+        return (this.pointer === 0);
+    }
+
+    /**
+     * Tell if current pointed element is last in history.
+     * It means we can't go any further backward
+     * It means pointer is on first snapshot taken
+     *
+     * @readOnly
+     * @return {Boolean}
+     */
+    isLast() {
+        return (this.pointer === this.snapshots.length - 1 && this.snapshots.length > 0);
+    }
+
+    /**
+     * Remove all snapshots from history.
+     * @return {undefined}
```

Comment:
The return {undefined} in JsDoc is unnecessary. .

Reply:
It was copied from data-store doc, if you remove it you can remove it in data-store too